server/server.go

```diff
@@ -177,9 +170,19 @@ func (server *Server[state]) ContextMiddleware() func(http.Handler) http.Handler
 	}
 }
 
+func noop(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		next.ServeHTTP(w, r)
+	})
+}
+
 // SessionMiddleware returns the Apollo session middleware.
 // Attach this before adding routes, if you want to use sessions.
 func (server *Server[state]) SessionMiddleware() func(http.Handler) http.Handler {
+	if server.sessionStore == nil {
+		slog.Warn("Not enabling the SessionMiddleware since there is no SessionStore configured")
+		return noop
+	}
 	return func(next http.Handler) http.Handler {
 		if server.sessionStore == nil {
```

Comment:
Is deze check niet overbodig nu?

Reply:
Die warning is gewoon om je daarop te wijzen als je dat per ongeluk vergeten zou zijn. Applicaties zonder user login hebben geen sessions nodig dus beide gevallen kunnen correct zijn

Reply:
Ik bedoel die eronder, met de panic

Reply:
Die zou in principe nooit getriggerd mogen worden, maar ik heb ze er in gelaten voor de zekerheid, dat is nog steeds duidelijker dan een nilpointer exception (maar ik wil ze er gerust ook uithalen, zou in principe niet mogen uitmaken)

Reply:
Ik zou het er uit halen, twee keer exact hetzelfde checken, meteen na elkaar lijkt me echt overbodig.

Reply:
Het is niet meteen na elkaar eh, de warning is wanneer je ze probeert te initialiseren en de panic is als je de middleware toch uitvoert zonder session store 😛

Reply:
Ah ja, had ik over gekeken. Nog steeds overbodig maar iets minder hahaha. Met die extra check is ook goed voor mij hoor

Reply:
Ik heb het er uit gehaald 😛